### PR TITLE
Caulaflower generic ammo

### PR DESCRIPTION
**Pull request by Caulaflower**
## Additions

Generic ammo mode, which groups calibers into generic calibers (like pistol ammo, intermediate ammo, etc.)

## To do
- translation file for settings [done]
- xml for generic ammo types [done]

### CODE REVIEW

File: Defs/Ammo/GENERIC/LauncherGrenade.xml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    <stackLimit>500</stackLimit>
+    <cookOffFlashScale>100</cookOffFlashScale>
+  </ThingDef>
+
+  <ThingDef Class="CombatExtended.AmmoDef" ParentName="AmmoLauncherGrenadeBase">
+    <defName>Ammo_LauncherGrenade_HE</defName>
+    <label>launcher grenade (HE)</label>
+    <graphicData>
+      <texPath>Things/Ammo/STACK_RANGED/GrenadeLauncher/HE</texPath>
+      <graphicClass>Graphic_StackCountRanged</graphicClass>
```
Comment (N7Huntsman) [inline Defs/Ammo/GENERIC/LauncherGrenade.xml:49]: These should be `CombatExtended.Graphic_StackCountRanged`.

File: Defs/Ammo/GENERIC/LauncherGrenade.xml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        </fragments>
+      </li>
+    </comps>
+  </ThingDef>
+
+  <ThingDef Class="CombatExtended.AmmoDef" ParentName="AmmoLauncherGrenadeBase">
+    <defName>Ammo_LauncherGrenade_Smoke</defName>
+    <label>launcher grenade (Smoke)</label>
+    <graphicData>
+      <texPath>Things/Ammo0/STACK_RANGED/GrenadeLauncher/SMK</texPath>
```
Comment (N7Huntsman) [inline Defs/Ammo/GENERIC/LauncherGrenade.xml:87]: Typo here. "Ammo0"

File: Defs/Ammo/GENERIC/LauncherGrenade.xml [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    <label>launcher grenade (HE)</label>
+    <graphicData>
+      <texPath>Things/Ammo/STACK_RANGED/GrenadeLauncher/HE</texPath>
+      <graphicClass>Graphic_StackCountRanged</graphicClass>
+    </graphicData>
+    <statBases>
+      <MarketValue>2.5</MarketValue>
+    </statBases>
+    <ammoClass>GrenadeHE</ammoClass>
+	<detonateProjectile>Bullet_LauncherGrenade_HE</detonateProjectile>
```
Comment (N7Huntsman) [inline Defs/Ammo/GENERIC/LauncherGrenade.xml:55]: This projectile doesn't exist. Should it be `Bullet_40x46mmGrenade_HE`?

**Review by N7Huntsman — DISMISSED** [review]
Another minor error:
```
Mod CETeam.CombatExtended-genericammo has multiple Verse.ThingCategoryDefs named AmmoHighCaliber. Skipping.
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
Verse.DefDatabase`1<Verse.ThingCategoryDef>:AddAllInMods ()
System.Reflection.MonoMethod:Invoke (object,System.Reflection.BindingFlags,System.Reflection.Binder,object[],System.Globalization.CultureInfo)
System.Reflection.MethodBase:Invoke (object,object[])
Verse.GenGeneric:InvokeStaticMethodOnGenericType (System.Type,System.Type,string)
Verse.PlayDataLoader:DoPlayLoad ()
Verse.PlayDataLoader:LoadAllPlayData (bool)
Verse.Root/<>c:<Start>b__6_1 ()
Verse.LongEventHandler:RunEventFromAnotherThread (System.Action)
Verse.LongEventHandler/<>c:<UpdateCurrentAsynchronousEvent>b__27_0 ()
System.Threading.ThreadHelper:ThreadStart_Context (object)
System.Threading.ExecutionContext:RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool)
System.Threading.ExecutionContext:Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool)
System.Threading.ExecutionContext:Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object)
System.Threading.ThreadHelper:ThreadStart ()
```

**Comment by N7Huntsman** [comment]
Regarding some other existing ammo options, with the implementation of this, should we remove the `Simplified Ammo` option?

**Comment by N7Huntsman** [comment]
Additionally, it currently doesn't swap out the ammo added in scenarios with their generic equivalent.

**Comment by Caulaflower** [comment]
> Regarding some other existing ammo options, with the implementation of this, should we remove the `Simplified Ammo` option?

in my opinion, there's no need to

**Comment by albertbokor** [comment]
> Regarding some other existing ammo options, with the implementation of this, should we remove the `Simplified Ammo` option?

ye, it's pretty useless

**Comment by albertbokor** [comment]
there was some fuckery with the high caliber rounds, renaming it to `AntiMateriel` to be less confusing

**Review by albertbokor — DISMISSED** [review]
works great now aside fromthat startup error

**Comment by perkinslr** [comment]
> Regarding some other existing ammo options, with the implementation of this, should we remove the `Simplified Ammo` option?

They do different things.  Simplified Ammo removes the APHE/AP-I listings, this groups calibers together but *doesn't* remove ammo types within the remaining calibers.  Some people may still want the extra challenge of no AP-I.

**Review by perkinslr — DISMISSED** [review]
Does what it says on the tin.  I am concerned about this decreasing the effective damage on some weapons that get their ammo swapped for something weaker, but it's up to the players to either accept that and work around it, or not use the setting.  Same as the no-ammo players.

**Review by albertbokor — DISMISSED** [review]
Works flawlessly as I've seen.

File: Source/CombatExtended/CombatExtended/Settings.cs [hunk 4]
```diff
@@ -14,6 +14,8 @@ public class Settings : ModSettings
         #region Settings
 
         // General settings
+        private bool bipodMechanics = true;
```
Comment (perkinslr) [inline Source/CombatExtended/CombatExtended/Settings.cs:17]: Bipods have their own PR (which was merged), why is this *here*?

**Review by perkinslr — CHANGES_REQUESTED** [review]

**Review by perkinslr — APPROVED** [review]

**Pull request merged**